packages/shopify-codemod/transforms/flatten-iife-global-assignment.js

```diff
@@ -0,0 +1,113 @@
+import {findFirstMember, getBlockStatementFromFunction} from './utils';
+
+export default function flattenIifeGlobalAssignment(
+  {source},
+  {jscodeshift: j},
+  {printOptions = {}, appGlobalIdentifiers},
+) {
+  const root = j(source);
+
+  function findAssignments() {
+    return root
+      .find(j.ExpressionStatement, {
+        expression: {
+          type: 'AssignmentExpression',
+          operator: '=',
+          left: {
+            type: 'MemberExpression',
+            object: (object) =>
+              appGlobalIdentifiers.indexOf(findFirstMember(object).name) >= 0,
+          },
+          right: {
+            type: 'CallExpression',
+            callee: {
+              type: 'FunctionExpression',
+            },
+          },
+        },
+      });
+  }
+
+  function nodeIsReturned(returnIdentifier, node) {
+    return node.type === 'FunctionDeclaration' && node.id.name === returnIdentifier;
+  }
+
+  function nodeIsAssignedToReturn(returnIdentifier, node) {
+    return (
+      returnIdentifier &&
+      node.type === 'ExpressionStatement' &&
+      findFirstMember(node.expression.left).name === returnIdentifier
+    );
+  }
+
+  function createIifeReturnAssignment(member, node) {
+    return j.expressionStatement(
+      j.assignmentExpression(
+        '=',
+        member,
+        j.functionExpression(member.property, node.params, node.body),
+      ),
+    );
+  }
+
+  function createIifeReturnMemberAssignment(member, node) {
+    return j.expressionStatement(
+      j.assignmentExpression(
+        node.expression.operator,
+        formatReturnAssignmentMemberExpression(node.expression.left, member),
+        node.expression.right,
+      ),
+    );
+  }
+
+  function formatReturnAssignmentMemberExpression(memberExpression, prependee) {
+    const identifiers = [];
+    let currentNode = memberExpression;
+
+    // Take all but the last, which is copied from the base memberExpression
+    while (currentNode.type === 'MemberExpression') {
```

Comment:
There's a `removeFirstMember` function in the `global-assignment-to-default-export` transforms.  That may be usable/adaptable here.